tests/acceptance/core/service/factory/cluster.go

```diff
@@ -0,0 +1,114 @@
+package factory
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"sync"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/k3s-io/k3s/tests/acceptance/shared/util"
+
+	. "github.com/onsi/ginkgo/v2"
+)
+
+var (
+	once    sync.Once
+	cluster *Cluster
+)
+
+type Cluster struct {
+	Status           string
+	ServerIPs        string
+	AgentIPs         string
```

Comment:
Since this is always a comma separated string, just store it as a slice from the beginning
```suggestion
	ServerIPs        []string
	AgentIPs         []string
```